.github/workflows/nix-shell-build.yml

```diff
@@ -0,0 +1,30 @@
+name: build and test Bitcoin Core with the nix-shell
+
+on:
+  workflow_dispatch:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  build:
+    strategy:
+      matrix:
+        nixos-version: [ "nixos-24.05", "nixos-unstable" ] 
+        bitcoin-ref: [ "master", "v27.1" ]
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v4
+    - name: Install Nix
+      uses: cachix/install-nix-action@v22
+      with:
+        nix_path: nixpkgs=channel:${{ matrix.nixos-version }}
+    - name: Checkout Bitcoin Core
+      uses: actions/checkout@v4
+      with:
+        repository: bitcoin/bitcoin
+        ref: ${{ matrix.bitcoin-ref }}
+        path: bitcoin
+    - name: Build with nix-shell (and aliases)
+      run: nix-shell --command "cd bitcoin && bash -c 'a && c && m && t'"
```

Comment:
hm yeah that won't work: `bash: line 1: a: command not found`

I've also tried without invoking bash - same problem.

see also: https://github.com/NixOS/nix/issues/1598

Reply:
Bash aliases in non-interactive shells seem generally to be a problem, even when setting `shopt -s expand_aliases`. Using bash functions should work both in interactive and non-interactive shells.

Reply:
Irritating that it seems to be working for some but not for others in the referenced nix issue. Feels a bit cleaner with aliases but I can live with the current compromise here for now. Thanks for trying to make it work.